### PR TITLE
Update core dependencies to v0.70.3

### PR DESCRIPTION
**Pull request by jessepinho**
## In this PR
- Update Cargo and NPM dependencies on core packages.
- Fix various places where we were previously using epochs, but are now using block heights.
  - This includes unbonding tokens. Unbonding tokens used to be named like `uunbonding_epoch_N_penumbravalid1abc123`. Now, unbonding tokens refer to a start _block height_, rather than a start _epoch_, so they're of the form `unbonding_start_at_N_penumbravalidabc123`. Also, they now have multiple denom units, so the display denom just starts with `unbonding_` rather than `uunbonding`.
- Fix a bug with how we stored epochs.
  - Previously, we stored new epochs only at epoch transitions. However, this meant that the first epoch (index 0) was missing! So, I updated the block processor to create an initial epoch if we're starting sync at block height 0.

### CODE REVIEW

File: packages/wasm/crate/Cargo.toml
Comment (grod220) [inline packages/wasm/crate/Cargo.toml:23]: These two packages do not seem to be used in our code, we can scrap

Comment (TalDerei) [reply packages/wasm/crate/Cargo.toml]: removed

File: apps/extension/.env
Comment (jessepinho) [inline apps/extension/.env:3]: Needed for the update to unbonding tokens, which now have a different symbol

File: packages/wasm/crate/Cargo.toml
Comment (jessepinho) [inline packages/wasm/crate/Cargo.toml:18]: Just updating `v0.68.3` to `v0.70.2` on all of these

File: packages/query/src/block-processor.ts
Comment (jessepinho) [inline packages/query/src/block-processor.ts:281]: Needed to `await` due to an unlikely, but possible, race condition. When we save a new epoch to the database during an epoch transition, we look in the database for the _previous_ latest known epoch. We grab its epoch index, increment it by 1, then use that index for the new epoch we're saving to the database.

If that took a long time, for some reason, and in the meantime the block processor had already gotten to the next epoch, we could have two epochs with the same epoch index.

This change makes sure that epoch indexes increment in the correct order.

File: packages/router/src/grpc/view-protocol-server/asset-metadata-by-id.test.ts
Comment (jessepinho) [inline packages/router/src/grpc/view-protocol-server/asset-metadata-by-id.test.ts:78]: There's really no need to throw, since it's acceptable to return an empty response from this RPC method. So I changed the behavior.

File: packages/storage/src/indexed-db/index.ts
Comment (jessepinho) [inline packages/storage/src/indexed-db/index.ts:461]: We're never actually using the `index` argument anywhere, so I just got rid of it.

File: packages/ui/components/ui/tx/view/undelegate.tsx
Comment (jessepinho) [inline packages/ui/components/ui/tx/view/undelegate.tsx:18]: This is a little strange to me: the `startEpochIndex` property is deprecated, so we should no longer show it. Ideally, we'd show the unbonding start height instead – but that property doesn't exist on the `Undelegate` proto! So for now, we'll just hide this, and revisit this if/when core adds that property to the `Undelegate` action.

File: packages/ui/components/ui/tx/view/value/index.tsx
Comment (jessepinho) [inline packages/ui/components/ui/tx/view/value/index.tsx:36]: Reviewers: I could use some help here.

When a user clicks the Undelegate button, the transaction approval dialog opens. We try to fetch the metadata for the undelegation tokens they'll be getting, but that will always come back empty, since the unbonding token type is now specific to the block height at which unbonding starts. That is, the unbonding token name is `unbonding_start_at_N_penumbravalid1...`, where `N` is the block height that unbonding started at.

There will _never_ be token metadata for an unbonding token until _after_ the transaction has been committed to the chain, since the token type is created when the transaction is committed.

But that means that the undelegation tokens are shown as "Unknown asset" in the transaction approval dialog:
image

Perhaps we should generate the unbonding token metadata on the fly here? If so, could someone point me in the right direction for how to do that? (That said, I'd love to get this PR merged before handling that, since _all_ delegation is blocked until this PR is merged.)

Comment (jessepinho) [reply packages/ui/components/ui/tx/view/value/index.tsx:36]: Another wrinkle to all this: this issue, where the metadata request always returns an empty response for an undelegation token that doesn't exist yet, wasn't happening before when undelegation tokens were tied to epochs rather than block heights.

I don't know how that's possible. We should have had the same problem back then, right? That is:

1. We transition to a new epoch.
2. Someone goes to undelegate tokens from a validator.
3. Since no one else has undelegated tokens from that validator in this epoch yet, the metadata for `uunbonding_epoch_N_...` doesn't exist yet, so the metadata response is empty.
4. `getDisplayDenomExponent` throws, because it's being passed `undefined` metadata.

And yet, this wasn't happening before — no error was being thrown. It only started with this change, where unbonding tokens are tied to block height rather than epoch index.

I'm investigating this, but it's a bit concerning since I don't understand why it worked before!

File: packages/wasm/crate/src/planner.rs
Comment (jessepinho) [inline packages/wasm/crate/src/planner.rs:310]: The epoch in the rate data has been deprecated, so we have to pass it in now.

File: packages/query/src/block-processor.ts
Comment (jessepinho) [inline packages/query/src/block-processor.ts:158]: Previously, we were skipping saving the first epoch to the database. This fixes that bug.

File: packages/router/src/grpc/view-protocol-server/asset-metadata-by-id.test.ts
Comment (grod220) [reply packages/router/src/grpc/view-protocol-server/asset-metadata-by-id.test.ts:78]: Response on other comment

File: packages/storage/src/indexed-db/index.ts
Comment (grod220) [inline packages/storage/src/indexed-db/index.ts:468]: Can we use an async iterator here?

```typescript
  async addEpoch(startHeight: bigint): Promise<void> {
    for await (const cursor of this.db.transaction('EPOCHS').store) {
      const previousEpoch = Epoch.fromJson(cursor.value);
      const index = previousEpoch.index + 1n;
      const newEpoch = {
        startHeight: startHeight.toString(),
        index: index.toString(),
      };
      await this.u.update({
        table: 'EPOCHS',
        value: newEpoch,
      });
    }
  }
```

Comment (grod220) [reply packages/storage/src/indexed-db/index.ts:468]: think getEpochByHeight() can be converted to one too

File: packages/ui/components/ui/tx/view/value/index.tsx
Comment (grod220) [reply packages/ui/components/ui/tx/view/value/index.tsx:36]: > Perhaps we should generate the unbonding token metadata on the fly here?

The `assetMetadataById` rpc method feels like a good place to generate metadata for this edge case so web apps have meaningful display data here.

> And yet, this wasn't happening before

Umm, no idea! 😅

File: packages/wasm/crate/src/storage.rs
Comment (grod220) [inline packages/wasm/crate/src/storage.rs:323]: Maybe we can combine these?

```rust
    Ok(store
        .open_cursor_with_direction(web_sys::IdbCursorDirection::Prev)?
        .await?
        .map(|cursor| serde_wasm_bindgen::from_value(cursor.value()).ok())
        .flatten())
}
```

**Review by grod220 — APPROVED** [review]
Wow, some really domain specific knowledge here 😅 . Good job! Left some comments.

File: packages/ui/components/ui/tx/view/undelegate.tsx
Comment (turbocrime) [reply packages/ui/components/ui/tx/view/undelegate.tsx:18]: can you leave the code here and just style it hidden or comment it out?

**Comment by TalDerei** [comment]
Are we waiting for the 0.71.0 point release later today?

**Review by TalDerei — APPROVED** [review]

File: packages/storage/src/indexed-db/index.ts
Comment (jessepinho) [reply packages/storage/src/indexed-db/index.ts:468]: Hm, why do it that way? I only want to grab the very last epoch from the table, not iterate over all of them. I guess I could just add a `break` or `return` statement at the end of the first run through the loop, but that feels a bit unidiomatic — people reading this code in the future will initially see it as a loop, and then realize the code is only running once.

Comment (jessepinho) [reply packages/storage/src/indexed-db/index.ts:468]: (I did update `getEpochByHeight` to use this syntax though, per your suggestion.)

File: packages/wasm/crate/src/storage.rs
Comment (jessepinho) [reply packages/wasm/crate/src/storage.rs]: Ahh, good call. Done.

File: packages/storage/src/indexed-db/index.ts
Comment (grod220) [reply packages/storage/src/indexed-db/index.ts:468]: Ah, didn't realize that was happening. Can you add a code comment? :)

Comment (grod220) [reply packages/storage/src/indexed-db/index.ts:468]: Oh wait there is one already lol whoops

**Pull request merged**
